### DISCUSSION TITLE
Converting NULL to string is unsupported

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#3177] [Q&A]
When I retrieve record using custom model, below error gives.
I copied my pb_data from server and run in local it works fine.

```
sql: Scan error on column index 22, name \"authType\": converting NULL to string is unsupported
```

**Comment by ganigeorgiev** [comment]
I don't understand what the above is. Please provide a little more context and/or share your code and what you are trying to do.

**Reply by Oggy4th** [reply]
```json
{
  "errorDetails": "Failed to execute FindOtaById() with otaId 'rfytntnub1mzj6x': failed to retrieve ota by ID 'rfytntnub1mzj6x': sql: Scan error on column index 22, name \"authType\": converting NULL to string is unsupported",
  "errorMessage": "Internal server error."
}
```

```go
var _ models.Model = (*Ota)(nil)

func (m *Ota) TableName() string {
	return "otas" // The name of a collection.
}

type (
	Ota struct {
		models.BaseModel

		Balance          float64                 `db:"balance" json:"balance"`
		Username         string                  `db:"username" json:"username"`
		LastWithdrawDate types.DateTime          `db:"lastWithdrawDate" json:"lastWithdrawDate"`
		LeftTime         int                     `db:"leftTime" json:"leftTime"`
		CloseTime        int                     `db:"closeTime" json:"closeTime"`
		TotalDeposit     float64                 `db:"totalDeposit" json:"totalDeposit"`
		Status           string                  `db:"status" json:"status"`
		OtaStatementIds  types.JsonArray[string] `db:"otaStatementIds" json:"otaStatementIds"`
		PassIds          types.JsonArray[string] `db:"passIds" json:"passIds"`
		TopUpIds         types.JsonArray[string] `db:"topUpIds" json:"topUpIds"`
		AuthType         string                  `db:"authType" json:"authType"`
		AuthId           string                  `db:"authId" json:"authId"`
		IssuerId         string                  `db:"issuerId" json:"issuerId"`
		BranchId         string                  `db:"branchId" json:"branchId"`
	}

func OtaQuery(dao *daos.Dao) *dbx.SelectQuery {
	return dao.ModelQuery(&Ota{})
}

func FindOtaById(dao *daos.Dao, otaId string) (*Ota, error) {
	ota := &Ota{}

	err := OtaQuery(dao).Where(dbx.HashExp{"id": otaId, "status": "active"}).Limit(1).One(ota)
	if err != nil {
		return nil, fmt.Errorf("failed to retrieve ota by ID '%s': %w", otaId, err)
	}

	return ota, nil
}
```

**Reply by ganigeorgiev** [reply]
Also note that, with exception of the `json` field, PocketBase doesn't support `null` field values.

Additionally, the above seems like a `database/sql` error, meaning that if you are trying to populate a null db value in a string struct field, then yes it won't work. You have to either change the field type to `*string` or use [`sql.NullString`](https://pkg.go.dev/database/sql#NullString).

**Reply by Oggy4th** [reply]
![Screenshot 2023-08-23 at 16 43 05](https://github.com/pocketbase/pocketbase/assets/44497354/9870aced-2293-480f-9f8f-d08e0faada72)

I thought when field is empty, pocketbase will convert it to empty string.

**Reply by ganigeorgiev** [reply]
@Oggy4th There are a lot of different factors that could affect the field value, but similar to your other questions you are not providing enough context and I'm not really sure how to help you.

- How was the record created in the first place? Was it with a custom model or the `Record` struct?

- What version of PocketBase do you use? There were several changes in the latest release that normalizes the field column definition (see https://github.com/pocketbase/pocketbase/issues/2992#issuecomment-1650279619) to prevent `null` value insertion in case a custom model was used.

**Reply by ganigeorgiev** [reply]
Based on your other question in https://github.com/pocketbase/pocketbase/discussions/3166 I guess that you are using v0.16.9.

Consider upgrading to v0.17+. It comes with a system migration that will normalize the previously `null` inserted values.

If upgrading doesn't fix your issue, please export your collections configuration as json (_Admin UI > Settings > Export collections_) I'll try to investigate it further.
